syntax = "proto3";                                                                                                                                                                                              
package Mutation_Detailed;

message detailed_mutation {
    int32 position = 1;
    int32 chrom_idx = 2;
    int32 par_mut_nuc = 3;
    int32 boundary1_tie = 4;
    int32 boundary2_flag = 5;
}

message mutation_list {
    repeated detailed_mutation mutation = 1;
}


message data {
    repeated mutation_list node_mutations = 1;
    repeated string chromosomes = 2; 
    repeated int32 ref_nuc=3;
    string newick=4;
    int64 internal_nodes_count=5;
}
